apps/explorer/src/ui/utils/ButtonOrLink.tsx

```diff
@@ -1,21 +1,37 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { type ComponentProps, forwardRef } from 'react';
+import { type ComponentProps, forwardRef, useMemo } from 'react';
 
 import { LinkWithQuery, type LinkProps } from './LinkWithQuery';
 
 export interface ButtonOrLinkProps
     extends Omit<
         Partial<LinkProps> & ComponentProps<'a'> & ComponentProps<'button'>,
         'ref'
-    > {}
+    > {
+    prefixIcon?: JSX.Element;
+    postfixIcon?: JSX.Element;
+}
 
 export const ButtonOrLink = forwardRef<
     HTMLAnchorElement | HTMLButtonElement,
     ButtonOrLinkProps
->(({ href, to, ...props }, ref: any) => {
-    // External link:
+>(({ href, to, prefixIcon, postfixIcon, children, ...props }, ref: any) => {
+    const childrenContent = useMemo(() => {
+        if (prefixIcon || postfixIcon) {
```

Comment:
Any harm in not doing this special casing and just returning it inline flex with gap by default?